### PR TITLE
Initial get on the device and store of config in the onos store

### PR DESCRIPTION
**Pull request by Andrea-Campanella**
This PR allows to get the existing device configuration and store it into ONOS. 
The request is done with and empty path that fetches the whole device configuration in `json` form. 
The` json` is then parsed and stored as initial configuration for that device in the `onos-config` store. 

Currently passes the `SetNetworkConfig` as a function to factory and synchroniser, this avoids circular dependencies in the functional code but provides the same issues on the `test`.

### CODE REVIEW

File: pkg/southbound/synchronizer/synchronizer.go [hunk 1]
```diff
@@ -44,6 +46,9 @@ var log = logging.GetLogger("southbound", "synchronizer")
 
 const matchOnIndex = `(\=.*?]).*?`
 
+type SetNetworkConfig func(targetUpdates map[string]devicechange.TypedValueMap,
```
Comment (SeanCondon) [inline pkg/southbound/synchronizer/synchronizer.go]: might need to declare this in some new low level package in `utils`, to make sure it has no dependencies other than `api`

File: pkg/manager/manager.go [hunk 2]
```diff
@@ -145,7 +145,8 @@ func (m *Manager) Run() {
 	go listenOnResponseChannel(m.SouthboundErrorChan, m)
 	//TODO we need to find a way to avoid passing down parameter but at the same time not hve circular dependecy sb-mgr
 	go synchronizer.Factory(m.TopoChannel, m.OperationalStateChannel, m.SouthboundErrorChan,
-		m.Dispatcher, m.ModelRegistry, m.OperationalStateCache, southbound.TargetGenerator, m.OperationalStateCacheLock, m.DeviceChangesStore)
+		m.Dispatcher, m.ModelRegistry, m.OperationalStateCache, southbound.TargetGenerator, m.OperationalStateCacheLock,
+		m.DeviceChangesStore, m.GetTargetConfig)
```
Comment (SeanCondon) [inline pkg/manager/manager.go]: should this be `m.SetTargetConfig`?

File: pkg/southbound/synchronizer/synchronizer.go [hunk 3]
```diff
@@ -121,6 +130,52 @@ func New(context context.Context,
...
+		Type:     gnmi.GetRequest_CONFIG,
+		Encoding: gnmi.Encoding_JSON,
+	}
+	configResponse, errGet := target.Get(context, getAllRequest)
+	if errGet != nil {
+		log.Error("Can't load configuration on device %s : %s", device.ID, errGet)
+		//TODO propagate
+	}
+	log.Infof("complete response of configurable parameters %s", configResponse)
+	//THis shoudl really be just one, iterating for better safety
```
Comment (ray-milkey) [inline pkg/southbound/synchronizer/synchronizer.go]: Minor typo - "This should"

File: pkg/southbound/synchronizer/factory.go [hunk 4]
```diff
@@ -57,13 +58,19 @@ func Factory(topoChannel <-chan *topodevice.ListResponse, opStateChan chan<- eve
 			} else {
 				mStateGetMode = modelregistry.GetStateMode(mPlugin.GetStateMode())
 			}
+			mReadWritePaths, ok := modelRegistry.ModelReadWritePaths[modelName]
+			if !ok {
+				log.Warnf("Cannot check for read write paths for target %s with %s because "+
```
Comment (ray-milkey) [inline pkg/southbound/synchronizer/factory.go]: Following this code as best as I can to the end, it looks to me like jsonpaths.CorrectJSONPaths will get a nil dereference if mReadWritePaths is nil. I think you have to set it to an empty map here.

**Comment by Andrea-Campanella** [comment]
closing since this will be completly redone after we have a sable environment after merging and testing @kuujo changes.

**Pull request closed without merging**